src/libs/extension-api/registry/extension.registry.ts

```diff
@@ -430,4 +430,40 @@ export class UmbExtensionRegistry<
 			distinctUntilChanged(extensionAndKindMatchArrayMemoization),
 		) as Observable<Array<ExtensionTypes>>;
 	}
+	
+	/**
+	 * Appends a new condition to an existing extension
+	 * Useful to add a condition for example the Save And Publish workspace action shipped by core
+	 * As overwriting the whole extension to simply add an extra condition is not ideal as it causes a lot of duplicated code
+	 * and could easily get out of sync from the CMS core implementation if a 3rd party dev was to try and overwrite it
+	 * @param alias {string} - The alias of the extension to append the condition to
+	 * @param newCondition {UmbConditionConfigBase} - The condition to append to the extension.
+	 */
+	appendCondition(alias: string, newCondition: UmbConditionConfigBase) {
+		const allExtensions = this._extensions.getValue();
+		const extensionToUpdate = allExtensions.find((ext) => ext.alias === alias) as ManifestWithDynamicConditions;
+
+		if(extensionToUpdate === undefined) {
+			console.error(`Extension with alias ${alias} not found`);
```

Comment:
I guess this solution could be fine, but ideally the code support late coming Extensions.

So the appending awaits the existence of the extension.

Reply:
I think this could be done just by making this method async, and then await a method that observes for a specific alias and resolves once its present. A method like this `whenAlias(alias: string): Promise<ManifestTypes>`

Reply:
@nielslyngsoe I will look into this Thurs/Fri when I switch gears for different clients

Reply:
@nielslyngsoe there is a new commit where I have tried to implement this from your notes - hence the highlighted code is outdated